### PR TITLE
tests: create temporary dir/files than /usr directory.

### LINKED ISSUE #4816: posix-tests are modifying /usr and / through tests if run as root
## Expected Behavior
Tests shouldn't be modifying operating system entities, this may lead to data loss on user system. 

## Current Behavior
We have some test code in posix_test.go which unnecessarily modifies operating system files, which is perhaps a data loss issue for any user.
```
$ git grep "/usr" cmd/posix_test.go
cmd/posix_test.go:              posix, err := newPosix("/usr")
cmd/posix_test.go:              posixStorage, err = newPosix("/usr")
cmd/posix_test.go:              posixStorage, err = newPosix("/usr")
cmd/posix_test.go:              posixStorage, err = newPosix("/usr")
cmd/posix_test.go:              posixStorage, err = newPosix("/usr")
cmd/posix_test.go:              posixStorage, err = newPosix("/usr")
```

These tests need to be fixed properly. 

## Possible Solution
```diff
        // TestPosix for permission denied.
        if runtime.GOOS != globalWindowsOSName {
+               posixStorage, path, err = newPosixTestSetup()
+               if err != nil {
+                       t.Fatalf("Unable to create posix test setup, %s", err)
+               }
+               defer os.RemoveAll(path)
+
                // Initialize posix storage layer for permission denied error.
-               posixStorage, err = newPosix("/usr")
+               posixStorage, err = newPosix(path)
                if err != nil {
-                       t.Fatalf("Unable to initialize posix, %s", err)
+                       t.Errorf("Unable to initialize posix, %s", err)
+               }
+
+               // Create a directory with perms to disallow reads.
+               if err = os.Mkdir(pathJoin(path, "bin"), 0200); err != nil {
+                       t.Fatalf("Unable to create directory with permissions, %s", err)
                }
 
-               if err = posixStorage.PrepareFile("bin", "yes", 16); !os.IsPermission(err) {
+               if err = posixStorage.PrepareFile("bin", "yes", 16); err != errFileAccessDenied {
                        t.Errorf("expected: Permission error, got: %s", err)
                }
        }
```

## Steps to Reproduce (for bugs)
Login as root and try to run tests using `go test`.

## Context
Cleanup.

### PR DESCRIPTION
**Pull request by balamurugana**
Fixes #4816

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/posix_test.go [hunk 1]
```diff
@@ -1298,8 +1362,22 @@ func TestPosixPrepareFile(t *testing.T) {
...
 	if runtime.GOOS != globalWindowsOSName {
+		var permDeniedDir string
+		if permDeniedDir, err = ioutil.TempDir(globalTestTmpDir, "minio-"); err != nil {
+			t.Fatalf("Unable to create temporary directory. %v", err)
+		}
+		defer os.RemoveAll(permDeniedDir)
+		if err = os.Mkdir(slashpath.Join(permDeniedDir, "bin"), 0400); err != nil {
+			t.Fatalf("Unable to create temporary directory %v. %v", slashpath.Join(permDeniedDir, "bin"), err)
+		}
+		if ioutil.WriteFile(slashpath.Join(permDeniedDir, "bin", "yes"), []byte(""), 0400); err != nil {
```
Comment (vadmeste) [inline cmd/posix_test.go]: `if err = ioutil.WriteFile(...`

File: cmd/posix_test.go [hunk 2]
```diff
@@ -697,14 +717,29 @@ func TestPosixPosixListDir(t *testing.T) {
...
```
Comment (vadmeste) [inline cmd/posix_test.go]: `if err = ioutil.WriteFile(`, err is missing

**Review by vadmeste — CHANGES_REQUESTED** [review]
LGTM but only minor comments are needed to be addressed

File: cmd/posix_test.go (hunk 2 above)
Comment (balamurugana) [reply cmd/posix_test.go]: Done

File: cmd/posix_test.go (hunk 1 above)
Comment (balamurugana) [reply cmd/posix_test.go]: Done

**Review by vadmeste — DISMISSED** [review]
LGTM

**Comment by harshavardhana** [comment]
PR seems to be failing travis-ci

 ```	
posix_test.go:729: Unable to create file /tmp/minio-371757886/bin/yes. open /tmp/minio-371757886/bin/yes: permission denied
```

**Review by donatello — DISMISSED** [review]
LGTM. Perhaps it would be less repetitive to put the perm denied dir and file creation into separate test helper.

**Comment by vadmeste** [comment]
By the way @bala, won't we have stale files after those tests, since we change dir permissions, probably os.RemoveAll() won't be able to cleanup, can you check that ?

**Comment by balamurugana** [comment]
@vadmeste Due to permission is set, `os.RemoveAll()` doesn't succeed.  Fixing it now.

**Comment by balamurugana** [comment]
@vadmeste the latest refresh fixed stale files

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by donatello — APPROVED** [review]

**Pull request merged**